architecture-docs/adr-003.md

```diff
@@ -114,56 +114,60 @@ of the project.
   }
   ```
   
-- A complete commit would be like this:
+- The complete block commit processing will be done as follows at the end:
   
   ```rust
   flush_staking(&mut kv_store!(self), block_height, staking_buffer);
   flush_kv(&mut self.kvdb, kv_buffer);  // Write to the disk atomically.
   ```
 
-### Chain node state changes
+### Chain Node State Changes
 
-- Remove `AccountStorage` from `ChainNodeApp`.
+- `AccountStorage` will be removed from `ChainNodeApp` (as the node storage will be in the dedicated `COL_TRIE_NODE` column).
 
-- In `ChainState`, change `account_root` to `last_account_block_height`, which records the last block height there's any
-  staking state modified.
+- In `ChainState`, `account_root` will be changed to `last_account_block_height` which records the last block height where any staking state was modified.
   
-  Use the block height to query history state directly.
+- In ABCIQuery, the block height from the request will be used to query historical states directly.
 
-### Staled nodes cleanup
+### Staled Trie Nodes Cleanup
 
-When node is modified, the old version is staled. The keys of staled nodes are stored in column `COL_STALED_NODE`, indexed by the block height it's staled since.
+When a node is modified, its old version is staled. The keys of staled trie nodes will be stored in the dedicated `COL_STALED_NODE` columned, indexed by the block height where they became staled.
 
-In the future, we might want to prune the nodes too old based on this information.
+Pruning of these staled trie nodes (e.g. in the context of validator operation where only the most recent version is needed)
+is outside of the scope of this ADR.
 
-### Implementation plan
+### Implementation Steps
 
-One PR for each steps:
+The work needed for this migration will be split into two PRs:
 
-- Add jellyfish module in `chain-storage`, which implements all the integration groundworks mentioned above.
-- Make the switch, and remove the starling based `AccountStorage`.
+1. A jellyfish module will be added in `chain-storage` (this module will implements all the integration groundwork mentioned above).
+2. The storage will be switched to use the jellyfish module and the starling-related code (`AccountStorage` etc.) will be removed.
 
 ## Status
 
-Proposed
+Accepted
 
 ## Consequences
 
 ### Positive
 
-- Better integration with buffer storage, single batch write operation for all the states when commit.
-- Clear tree structure and argorithm.
-- No need to store the root hash in chain state, and the block height to root hash mapping.
+- Better integration with the existing buffer storage abstraction; all Chain node state modifications can be done in a  single batch write operation during the BlockCommit request processing
+- More robust trie data structure and algorithms
+- No need to store the root hash in the Chain node state, and the block height to root hash mapping
```

Comment:
```suggestion
- No need to store the block height to root hash mapping
```

It just comes to me that, the root hash in chain state is still useful to client to verify Merkel proof.